e2e/tests/playwright-tests/helpers/operations.ts

```diff
@@ -0,0 +1,97 @@
+    await page.waitForSelector(gmailSelectors.loginView.passwordInput, { state: 'visible', timeout: 30000 });
+    await page.locator(gmailSelectors.loginView.passwordInput).fill(userObject.password);
+    await page.locator(gmailSelectors.loginView.passwordNextBtn).click();
+
+    const selector = selectorAwaited === undefined ? gmailSelectors.composeBtn : selectorAwaited;
+    const { recoveryEmailItem } = gmailSelectors.verifyView;
+    const { notNowBtn } = gmailSelectors.verifyView;
+    await page.waitForSelector(`:is(${selector}, ${recoveryEmailItem}, ${notNowBtn})`, { timeout: 60000 });
+    if (await page.isVisible(recoveryEmailItem)) {
+        await page.locator(recoveryEmailItem).click();
+        await page.locator(gmailSelectors.verifyView.recoveryEmailInput).fill(userObject.recoveryEmail);
+        await page.locator(gmailSelectors.verifyView.nextBtn).click();
+        await page.locator(selector).waitFor({ state: 'visible', timeout: 60000 });
+    }
+    if (await page.isVisible(notNowBtn)) await page.locator(notNowBtn).click();
+
+    await page.goto('/');
+
+    await page.waitForSelector(selectors.firstScreen.unsupportedModal.continueAnywayBtn);
+    await page.locator(selectors.firstScreen.unsupportedModal.continueAnywayBtn).click();
+
+    await page.locator(selectors.loginButton).click();
+    await page.locator(selectors.loginGoogleButton).click();
+
+    await page.waitForURL('https://local.virtru.com/');
+    await page.waitForSelector(selectors.logoutButton);
+}
+
+type EncryptDecryptProcedure = {
+    page: Page,
+    recipientEmail?: string,
+    full?: boolean
+}
+
+export const encryptDecryptProcedure = async ({ page, recipientEmail, full }: EncryptDecryptProcedure) => {
+    await page.locator(selectors.draggableItem).dragTo(page.locator(selectors.dropZone));
+
+    if (recipientEmail) {
+        await page.fill('input[type=email]', recipientEmail);
+        await page.getByText(selectors.grantBtnText).click();
+        await expect(page.getByText(selectors.revokeBtnText)).toBeVisible();
+    }
+
+    await page.getByText(selectors.protectFileBtnText).click();
+
+    if (full) {
+        const responsePublicKey = await page.waitForResponse('**/auth/oidc/public-key');
+        await expect(responsePublicKey.status()).toEqual(200);
+
+        const responseToken = await page.waitForResponse('**/oauth2/default/v1/token');
+        await expect(responseToken.status()).toEqual(200);
+
+        const responseEntityObject = await page.waitForResponse('**/accounts/api/entityobject');
+        await expect(responseEntityObject.status()).toEqual(200);
+
+        const responseUserSettings = await page.waitForResponse('**/accounts/api/userSettings');
+        await expect(responseUserSettings.status()).toEqual(200);
+
+        const responseUpsert = await page.waitForResponse('**/kas/upsert');
+        await expect(responseUpsert.status()).toEqual(200);
```

Comment:
Shouldn't these be true for any call to 'protect'?

An alternative here is to make two separate functions, `protect` and `download`. It seems like it would be more clear than this, which is called `encryptAndDecrypt` but only decrypts if you pass the `full` parameter. Alternatively, rename to `encrypt` and rename `full` to `andDecrypt` :-)

```suggestion
    const responsePublicKey = await page.waitForResponse('**/auth/oidc/public-key');
    await expect(responsePublicKey.status()).toEqual(200);

    const responseToken = await page.waitForResponse('**/oauth2/default/v1/token');
    await expect(responseToken.status()).toEqual(200);

    const responseEntityObject = await page.waitForResponse('**/accounts/api/entityobject');
    await expect(responseEntityObject.status()).toEqual(200);

    const responseUserSettings = await page.waitForResponse('**/accounts/api/userSettings');
    await expect(responseUserSettings.status()).toEqual(200);

    const responseUpsert = await page.waitForResponse('**/kas/upsert');
    await expect(responseUpsert.status()).toEqual(200);
    if (full) {
```

Reply:
good point